Microsoft.Azure.Cosmos/src/CosmosElements/CosmosElement.cs

```diff
@@ -187,7 +189,32 @@ public static bool TryCreateFromBuffer<TCosmosElement>(ReadOnlyMemory<byte> buff
         public static CosmosElement Parse(string json)
         {
             TryCatch<CosmosElement> tryParse = CosmosElement.Monadic.Parse(json);
-            tryParse.ThrowIfFailed();
+            try
+            {
+                tryParse.ThrowIfFailed();
+            }
+            catch (Exception exception)
```

Comment:
```suggestion
            catch (Exception exception) when (exception.InnerException is JsonParseException)
```

Removes the need for the if/else

Reply:
With this change you would replace the if else with two catch blocks. The first block would be catch (Exception exception) when (exception.InnerException is JsonParseException) and the second would be catch(Exception ex) where we would throw the exception. 

Instead of having these two catch blocks I think its better to have the if/else we currently have.